### PR TITLE
Generator page changes

### LINKED ISSUE #501: Replace lodash with simpler templating tool
Lodash is only be used for the generator list to make each item and show the list. Would be nice if it could be replaced with something that is smaller.

### LINKED ISSUE #518: Document recommended way of publishing generator forks
Regarding https://github.com/yeoman/yeoman/issues/1533#issuecomment-132450964.

Forks meaning generators based on existing ones but with personal tweaks.

[Scoped packages](https://docs.npmjs.com/getting-started/scoped-packages) is the way to go as everyone can have the `generator-angular` name if they want with `@sindresorhus/generator-angular`.

### PR DESCRIPTION
**Pull request by eddiemonge**
- Add better instructions for adding a generator
 - Swap lodash for doT.js for templating
 - Update HTML to match doT.js syntax
 - Update list.js
 - Use new generator list location
 - Update syntax for generator list location
 - Fix date display and sorting
 - Removed the blacklist as it is in the yeoman-generator-list now

Fixes #501
Fixes #518

### CODE REVIEW

File: app/assets/js/generator.js [hunk 1]
```diff
@@ -2,53 +2,24 @@
...
-    return Math.floor(seconds) + ' seconds';
-  }
-
   $(function() {
-    // Use vps until heroku issues are resolved
-    $.getJSON('https://storage.googleapis.com/yeoman-generator-list/cache.json')
-    // $.getJSON('https://yeoman-generator-list.herokuapp.com')
+    var $instructions = $('#instructions').on('click', function (e) {
+      e.preventDefault();
+      $instructions.next().toggle('fast');
```
Comment (SBoudrias) [inline app/assets/js/generator.js]: Can we be explicit about the selector here?

File: app/assets/js/generator.js [hunk 2]
```diff
@@ -2,53 +2,24 @@
...
-        return el;
-      }).sort(function (a, b) {
+      var modules = plugins.sort(function (a, b) {
         return a.stars === b.stars ? 0 : a.stars < b.stars ? 1 : -1;
       });
 
-      var allTpl = _.template($('#plugins-all-template').html(), {
-        modules: modules
-      });
+      doT.templateSettings.interpolate = /\<\%=([\s\S]+?)\%\>/g;
```
Comment (SBoudrias) [inline app/assets/js/generator.js]: This should be set outside the callback

File: app/assets/js/generator.js [hunk 3]
```diff
@@ -2,53 +2,24 @@
...
+      doT.templateSettings.conditional = /\<\%\?(\?)?\s*([\s\S]*?)\s*\%\>/g;
+      doT.templateSettings.iterate = /\<\%~\s*(?:\%\>|([\s\S]+?)\s*\:\s*([\w$]+)\s*(?:\:\s*([\w$]+))?\s*\%\>)/g;
+
+      var tpl = doT.template($('#plugins-all-template').text());
```
Comment (SBoudrias) [inline app/assets/js/generator.js]: Let's not compile a template each time we fetch json data.

**Comment by SBoudrias** [comment]
Why `doT.js`? I'm not sure I like the syntax, just reading through the template now, I can't understand what's it doing.

**Comment by eddiemonge** [comment]
It came up in a search and was super tiny. It is a bit complicated to work with and its default `{{ }}` kept getting stripped out by I think Jekyll. I'm not attached to it. Didn't feel like having to build lodash.template everytime there was updates as there doesn't seem to be a prebuilt stand-alone.

File: app/generators/index.html [hunk 4]
```diff
@@ -5,7 +5,27 @@
...
   <section class="main">
-    <small>Your generator must have a GitHub repository description, the `yeoman-generator` keyword and a description in `package.json` to be listed. Official generators are marked with <img src="/assets/img/mustache.svg" width="20px"></small>
+    <a id="instructions-toggle">How to add your generator &raquo;</a>
+    <div id="instructions" style="display:none">
+      <p>
+        Your generator will automatically show up in the list if it meets the following criteria. The list is populated every hour.
+      </p>
+      <ul>
+        <li>Be published on the npmjs.org repository</li>
+        <li>Have the <code>yeoman-generator</code> keyword</li>
```
Comment (sindresorhus) [inline app/generators/index.html]:
```
<li>Have the <code>yeoman-generator</code> keyword in package.json</li>
```

File: app/generators/index.html [hunk 5]
```diff
@@ -5,7 +5,27 @@
...
+        <li>Have a description in the <code>package.json</code> file or in its associated Github repository</li>
```
Comment (sindresorhus) [inline app/generators/index.html]: Drop the mention of GitHub description. We support package.json description now, so we should just prefer that.

**Comment by sindresorhus** [comment]
> there doesn't seem to be a prebuilt stand-alone.

https://www.npmjs.com/package/lodash.template

**Comment by eddiemonge** [comment]
> there doesn't seem to be a prebuilt stand-alone.

> >  https://www.npmjs.com/package/lodash.template

Which is fine for node but requires a lot more things in the browser:
https://github.com/lodash/lodash/blob/3.6.2-npm-packages/lodash.template/package.json#L20-L28

**Comment by sindresorhus** [comment]
We should just use Browserify or Webpack for the website, but I guess that's another PR.

**Comment by SBoudrias** [comment]
I'm not sure about either of these, we have a very lightweight system ATM. Only the search page is using some very light jQuery. I'm not sure this is worth it.

**Pull request merged**